google/cloud/bigquery/_helpers.py

```diff
@@ -382,7 +378,14 @@ def _field_to_index_mapping(schema):
 
 
 def _field_from_json(resource, field):
-    converter = _CELLDATA_FROM_JSON.get(field.field_type, lambda value, _: value)
+    def default_converter(value, field):
+        warnings.warn(
+            "Unknown type '{}' for field '{}'.".format(field.field_type, field.name),
```

Comment:
Let's be explicit and say "Behavior reading this type is not officially supported and may change in future." or something scary along those lines.

Reply:
Before we mark this as "fixed", I think the inverse direction is needed too (where we write JSON from a record), which is used in methods like `client.insert_rows()` (though maybe that already fails when encountering unknown types?)

Reply:
Got it. Yeah looks like https://github.com/googleapis/python-bigquery/blob/4383cfe7c7571ffaa3efd5e45ca33c6ff978f274/google/cloud/bigquery/_helpers.py#L690 and https://github.com/googleapis/python-bigquery/blob/4383cfe7c7571ffaa3efd5e45ca33c6ff978f274/google/cloud/bigquery/_helpers.py#L624 are two points. The first there is no type info so probably just _scalar_field_to_json needs to be updated. Updated PR to include a warning for the write.